frontend/online-shop/src/App.tsx

```diff
@@ -0,0 +1,26 @@
+import "./App.css";
+import { TableProducts } from "./components/Products";
+import { Switch, Route, Redirect } from "react-router-dom";
+import ProductDetails from "./components/ProductDetails";
+import { ThemeProvider } from "@material-ui/core";
+import theme from "./styles/tableTheme";
+import Header from "./Header";
+
+export const App = () => {
+  return (
+    <ThemeProvider theme={theme}>
+      <div>
+        <Header />
+        <main>
+          <Switch>
```

Comment:
What if the user is directly accessing a path that you don't have in your switch? For example http://localhost:4200/asdfg. For such cases you need to have a wildcard route that catches all the paths that do not fit with any other declared route and return a 404 page.

Reply:
Refactored to use only Material UI